### DISCUSSION TITLE
A Tour of the Composable Architecture: Part 3 - List does not animate

### DISCUSSION THREAD

**Discussion by jmgawecki** [pointfreeco/swift-composable-architecture#553] [Q&A] [answered]
Hello everyone, 
I am following A Tour of the Composable Architecture, I am currently on Part 3. In this part, where implementing a sorting mechanism, todos are sorting nicely. But they are sorting with the animation according to the video, whereas, my todos are sorting with no animation at all, instantly. I've tried to wrap some elements in Animation blocks but it didn't change a thing. 

I am curious, why my code, which is identical (if I haven't made any mistake) to the one from the tutorial, works differently. 

Many thanks for any suggestions. 

Here is my code:
https://github.com/jmgawecki/TodoComposableArchitectureTutorial

**Comment by mbrandonw** [comment, marked as answer]
Hi @jmgawecki, thanks for the question. This is actually due to a change in SwiftUI between iOS 13 and 14. In iOS 13, SwiftUI would implicitly animate a lot of things for you (such as rows in a `List`) without you needing to do anything in code. But in iOS 14 SwiftUI started requiring you to be more explicit with animations.

There are a few ways you can get animation back. You could tack on a `.animation` view modifier to the list:

```swift
List {
  ForEachStore(
    self.store.scope(state: \.todos, action: AppAction.todo(index:action:)),
    content: TodoView.init(store:)
  )
}
.animation()
```

This will make it so that the `List` always animates its contents.

Alternatively, you can be a little bit more focused when the list animates by using animated schedulers, which is a concept we discussed in [this](https://www.pointfree.co/collections/swiftui/animations) collection of episodes. You can do this by using the `.animation` method on the `Scheduler` protocol to transform `DispatchQueue.main` into a scheduler that animates its work:

```swift
return Effect(value: AppAction.todoDelayCompleted)
  .delay(for: 1, scheduler: /*➡️*/ DispatchQueue.main.animation() /*⬅️*/)
  .eraseToEffect()
  .cancellable(id: CancelDelayId(), cancelInFlight: true)
```

We should probably add a correction to the episode to explain this behavior to our viewers. We'll try to do that soon.

**Reply by jmgawecki** [reply]
Hey Brandon,
Thank you for getting back to me. I will definitely dive into the collection of those episodes later.
Clearly, the issue is not related to Composable Architecture, and yet You help me and that I appreciate:))
